### PR TITLE
Extend closed issues job to check issues in other repos

### PR DESCRIPTION
**Pull request by strkkk**
Related to #9256 

1. Job is changed to detect closed issues in other repos as well
2. Violations are fixed (spotbugs and pitest)

### CODE REVIEW

**Comment by strkkk** [comment]
Azure job failed because of connection issue

**Comment by strkkk** [comment]
@pbludov Builds on java11+ jdks are failing on spotbugs
```
[INFO] --- spotbugs-maven-plugin:4.2.0:check (default) @ checkstyle ---
[INFO] BugInstance size is 7
[INFO] Error size is 0
[INFO] Total bugs: 7
[ERROR] Low: Redundant nullcheck of stream, which is known to be non-null in com.puppycrawl.tools.checkstyle.Main.loadProperties(File) [com.puppycrawl.tools.checkstyle.Main] Redundant null check at Main.java:[line 431] RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE
[ERROR] Low: Redundant nullcheck of inStream, which is known to be non-null in com.puppycrawl.tools.checkstyle.PropertyCacheFile.load() [com.puppycrawl.tools.checkstyle.PropertyCacheFile] Redundant null check at PropertyCacheFile.java:[line 125] RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE
[ERROR] Low: Redundant nullcheck of out, which is known to be non-null in com.puppycrawl.tools.checkstyle.PropertyCacheFile.persist() [com.puppycrawl.tools.checkstyle.PropertyCacheFile] Redundant null check at PropertyCacheFile.java:[line 151] RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE
[ERROR] Low: Redundant nullcheck of inStream, which is known to be non-null in com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask.createOverridingProperties() [com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask] Redundant null check at CheckstyleAntTask.java:[line 441] RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE
[ERROR] Low: Redundant nullcheck of inputStream, which is known to be non-null in com.puppycrawl.tools.checkstyle.checks.OrderedPropertiesCheck.processFiltered(File, FileText) [com.puppycrawl.tools.checkstyle.checks.OrderedPropertiesCheck] Redundant null check at OrderedPropertiesCheck.java:[line 131] RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE
[ERROR] Low: Redundant nullcheck of inStream, which is known to be non-null in com.puppycrawl.tools.checkstyle.checks.TranslationCheck.getTranslationKeys(File) [com.puppycrawl.tools.checkstyle.checks.TranslationCheck] Redundant null check at TranslationCheck.java:[line 598] RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE
[ERROR] Low: Redundant nullcheck of inputStream, which is known to be non-null in com.puppycrawl.tools.checkstyle.checks.UniquePropertiesCheck.processFiltered(File, FileText) [com.puppycrawl.tools.checkstyle.checks.UniquePropertiesCheck] Redundant null check at UniquePropertiesCheck.java:[line 136] RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE
[INFO]
```
Is it related to https://github.com/checkstyle/checkstyle/issues/9297? 
Looks weird because it passes on java8

**Review by romani — CHANGES_REQUESTED** [review]
I am not firm on this but I think we should move this validation to another CI.
Example:
https://app.codeship.com/projects/124310/builds/dda430a4-0442-4e8f-a3d6-09450bad79c9?step=parallel_ls_-la
main reason is not that we already do link validation in this CI :) it is just coincident.

main reason is that this validation will fail by any reason that  is not related to PR code changes, so there will be bunch of failures in CI and we always would need to recheck, contributors get confused, ....
We need this validation in after merge, codeship exactly for this. All problems with stale links does not bother PRs, and maintainers eventually fix them.

**Comment by strkkk** [comment]
@romani do you mean move whole job to codeship or only check of external repos?

There is one thing - is there auth token in codeship to make requests to github? It is there in travis(READ_ONLY_TOKEN), but I am not sure about codeship

**Comment by romani** [comment]
I think we need to move whole job as even our own issue are closed after PR is merged.

We can try https://docs.cloudbees.com/docs/cloudbees-codeship/latest/pro-builds-and-configuration/environment-variables#_encrypted_environment_variables

**Review by romani — APPROVED** [review]

**Comment by strkkk** [comment]
I think I will try to move it to github actions to trigger it after merge (or by cron) instead of dealing with configuration of variables. READ_ONLY_TOKEN is same thing as GITHUB_TOKEN in github actions, isn't it?

Will try it as next step for this task.

**Comment by romani** [comment]
> READ_ONLY_TOKEN is same thing as GITHUB_TOKEN in github actions, isn't it?

yes.

File: .ci/no_old_refs.sh
Comment (pbludov) [inline .ci/no_old_refs.sh:11]: The pattern `[a-z]` is not enough here. There may be dashes and numbers in the uri, for example:
```
https://github.com/junit-team/junit5/issues/1786
https://github.com/Azure/actions/issues/2
https://github.com/Abhishek-kumar09/susi_server/1
https://github.com/strkkk/strkkk.github.io/issues/1
```

**Review by pbludov — CHANGES_REQUESTED** [review]
Please update the pattern.

File: .ci/no_old_refs.sh
Comment (strkkk) [reply .ci/no_old_refs.sh]: Good catch, fixed.
By the way, all mentioned repos in code have only lowercase :)

**Review by pbludov — APPROVED** [review]

**Pull request merged**
